File: frontend/src/services/modules/publish.js
import { post, get, put, del } from '../api.js';

export const publishApi = {
  // 获取产品列表
  getProducts: (params) => get('/publish/products', params),
  
  // 添加产品
  addProduct: (data) => post('/publish/products', data),
  
  // 更新产品
  updateProduct: (productId, data) => put(`/publish/products/${productId}`, data),
  
  // 删除产品
  deleteProduct: (productId) => del(`/publish/products/${productId}`),
  
  // 获取平台推荐
  getPlatformRecommendation: (data) => post('/publish/platform-recommendation', data),
  
  // 获取平台列表
  getPlatforms: () => get('/publish/platforms'),
  
  // 一键上架
  publish: (data) => post('/publish/publish', data),
  
  // 获取上架任务状态
  getPublishTask: (publishId) => get(`/publish/tasks/${publishId}`),
};

File: app/api/v1/publish.py
"""上架发布接口。"""
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_user_or_dev
from app.core.response import ok
from app.database import get_db
from app.models.user import User
from app.services import publish_service

router = APIRouter()


@router.get("/products")
def list_products(page: int = 1, page_size: int = 20, db: Session = Depends(get_db), user: User = Depends(get_current_user_or_dev)):
    return ok(publish_service.list_products(db, user, page, page_size))


@router.post("/products")
def create_product(body: dict, db: Session = Depends(get_db), user: User = Depends(get_current_user_or_dev)):
    return ok(publish_service.create_product(db, user, body))


@router.put("/products/{product_id}")
def update_product(product_id: str, body: dict, db: Session = Depends(get_db), user: User = Depends(get_current_user_or_dev)):
    return ok(publish_service.update_product(db, user, product_id, body))


@router.delete("/products/{product_id}")
def delete_product(product_id: str, db: Session = Depends(get_db), user: User = Depends(get_current_user_or_dev)):
    publish_service.delete_product(db, user, product_id)
    return ok(message="删除成功")


@router.post("/platform-recommendation")
def platform_recommendation(
    body: dict,
    db: Session = Depends(get_db),
    user: User = Depends(get_current_user_or_dev),
):
    return ok(publish_service.platform_recommendation(db, user, body))


@router.get("/platforms")
def platforms(user: User = Depends(get_current_user_or_dev)):
    return ok(publish_service.list_platforms())


@router.post("/publish")
def publish(body: dict, db: Session = Depends(get_db), user: User = Depends(get_current_user_or_dev)):
    return ok(publish_service.publish_product(db, user, body))


@router.get("/tasks/{publish_id}")
def publish_task(
    publish_id: str,
    db: Session = Depends(get_db),
    user: User = Depends(get_current_user_or_dev),
):
    return ok(publish_service.get_publish_task(db, user, publish_id))
